datavault/backend.py

```diff
@@ -51,24 +91,205 @@ def size(self):
+
+        for i, ind in enumerate(self.independents):
+            sec = 'Independent {}'.format(i+1)
+            S.add_section(sec)
+            S.set(sec, 'Label', ind.label)
+            S.set(sec, 'Units', ind.unit)
+
+        for i, dep in enumerate(self.dependents):
+            sec = 'Dependent {}'.format(i+1)
+            S.add_section(sec)
+            S.set(sec, 'Label',    dep.legend)
+            S.set(sec, 'Units',    dep.unit)
+            S.set(sec, 'Category', dep.label)
+
+        for i, par in enumerate(self.parameters):
+            sec = 'Parameter {}'.format(i+1)
+            S.add_section(sec)
+            S.set(sec, 'Label', par['label'])
+            # encode the parameter value as a data-url
+            data_url = labrad_urlencode(par['data'])
+            S.set(sec, 'Data', data_url)
+
+        sec = 'Comments'
+        S.add_section(sec)
+        for i, (time, user, comment) in enumerate(self.comments):
+            time = time_to_str(time)
+            S.set(sec, 'c{}'.format(i), repr((time, user, comment)))
+
+        with open(self.infofile, 'w') as f:
+            S.write(f)
+
+    def initialize_info(self, title, indep, dep):
+        self.title = title
+        self.accessed = self.modified = self.created = datetime.datetime.now()
+        self.independents = indep
+        self.dependents = dep
+        self.parameters = []
+        self.comments = []
+        self.cols = len(indep) + len(dep)
+
+    @property
+    def dtype(self):
+        return np.dtype(','.join(['f8']*self.cols))
+
+    def access(self):
+        self.accessed = datetime.datetime.now()
+
+    def getIndependents(self):
+        return self.independents
+
+    def getDependents(self):
+        return self.dependents
+
+    def getRowType(self):
+        units = []
+        for var in self.independents + self.dependents:
+            units.append('v[{}]'.format(var.unit))
+        type_tag = '*({})'.format(','.join(units))
+        return type_tag
+
+    def getTransposeType(self):
+        units = []
+        for var in self.independents + self.dependents:
+            units.append('*v[{}]'.format(var.unit))
+        type_tag = '({})'.format(','.join(units))
+        return type_tag
+
+    def addParam(self, name, data):
+        for p in self.parameters:
+            if p['label'] == name:
+                raise errors.ParameterInUseError(name)
+        d = dict(label=name, data=data)
+        self.parameters.append(d)
+
+    def getParameter(self, name, case_sensitive=True):
+        for p in self.parameters:
+            if case_sensitive:
+                if p['label'] == name:
+                    return p['data']
+            else:
+                if p['label'].lower() == name.lower():
+                    return p['data']
+        raise errors.BadParameterError(name)
+
+    def getParamNames(self):
+        return [p['label'] for p in self.parameters]
+
+    def addComment(self, user, comment):
```

Comment:
similar to the addParam case, we are relying on the labrad server frontend to pass appropriate types in here. But you're right that we should make this robust on its own.